### PR TITLE
gdal: enable tests

### PR DESCRIPTION
**Pull request by risicle**
###### Description of changes

Enabled tests. Tests are good.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/gdal/default.nix [hunk 1]
```diff
@@ -86,6 +86,41 @@ stdenv.mkDerivation rec {
...
+  # calls (coming from the python world)
+  preCheck = ''
+    pushd ../autotest
+    # something has made files here read-only by this point
+    chmod -R u+w .
+
+    export HOME=$(mktemp -d)
+    export PYTHONPATH="$out/${pythonPackages.python.sitePackages}:$PYTHONPATH"
+  '';
+  installCheckInputs = with pythonPackages; [
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gdal/default.nix:100]: Is that still python2?

Comment (risicle) [reply pkgs/development/libraries/gdal/default.nix:100]: No. Looks like it gets overridden when this is included in a `pythonPackages`.

It's not something I've changed anyway, and not something I particularly want to go screwing with.

**Comment by risicle** [comment]
I suspect a port 8080 collision on ofborg's darwin x86_64 run...

@ofborg build python39Packages.gdal python310Packages.gdal

**Review by Mindavi — APPROVED** [review]
LGTM, but didn't run the tests myself

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 166012` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>gplates</li>
    <li>merkaartor</li>
    <li>openorienteering-mapper</li>
  </ul>
</details>
<details>
  <summary>30 packages failed to build:</summary>
  <ul>
    <li>gdal (python39Packages.gdal)</li>
    <li>grass</li>
    <li>libLAS</li>
    <li>mapcache</li>
    <li>mapnik</li>
    <li>mapproxy</li>
    <li>mapserver</li>
    <li>perl532Packages.Tirex</li>
    <li>perl534Packages.Tirex</li>
    <li>postgresql11Packages.postgis</li>
    <li>postgresql12Packages.postgis</li>
    <li>postgresql13Packages.postgis</li>
    <li>postgresql14Packages.postgis</li>
    <li>python310Packages.cartopy</li>
    <li>python310Packages.fiona</li>
    <li>python310Packages.gdal</li>
    <li>python310Packages.geopandas</li>
    <li>python310Packages.osmnx</li>
    <li>python310Packages.python-mapnik</li>
    <li>python310Packages.rasterio</li>
    <li>python310Packages.worldengine</li>
    <li>python39Packages.cartopy</li>
    <li>python39Packages.fiona</li>
    <li>python39Packages.geopandas</li>
    <li>python39Packages.osmnx</li>
    <li>python39Packages.python-mapnik</li>
    <li>python39Packages.rasterio</li>
    <li>python39Packages.worldengine</li>
    <li>sumo</li>
    <li>t-rex</li>
  </ul>
</details>

**Comment by alexvorobiev** [comment]
@risicle Some tests still fail:
```
=========================== short test summary info ============================
FAILED gdrivers/wcs.py::test_wcs_1 - socket.timeout: timed out
FAILED gdrivers/wcs.py::test_wcs_3 - AttributeError: module 'gdaltest' has no...
FAILED gdrivers/wcs.py::test_wcs_4 - AttributeError: module 'gdaltest' has no...
= 3 failed, 7900 passed, 1499 skipped, 3 deselected, 2 xfailed, 25 warnings in 348.59s (0:05:48) =
100 - done.
error: builder for '/nix/store/53wpra5lmfjnbcw26y41p8xj95dnhfz6-gdal-3.4.2.drv' failed with exit code 1;
       last 10 log lines:
       >   /nix/store/xqacz3jw91cgbbjzdi2syd974lp1yrq4-gdal-3.4.2/lib/python3.9/site-packages/osgeo/gdal.py:1528: DeprecationWarning: ApplyVerticalShiftGrid()
will be removed in GDAL 4.0
       >     warn('ApplyVerticalShiftGrid() will be removed in GDAL 4.0', DeprecationWarning)
       >
       > -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
       > =========================== short test summary info ============================
       > FAILED gdrivers/wcs.py::test_wcs_1 - socket.timeout: timed out
       > FAILED gdrivers/wcs.py::test_wcs_3 - AttributeError: module 'gdaltest' has no...
       > FAILED gdrivers/wcs.py::test_wcs_4 - AttributeError: module 'gdaltest' has no...
       > = 3 failed, 7900 passed, 1499 skipped, 3 deselected, 2 xfailed, 25 warnings in 348.59s (0:05:48) =
       > 100 - done.
       For full logs, run 'nix log /nix/store/53wpra5lmfjnbcw26y41p8xj95dnhfz6-gdal-3.4.2.drv'.
error: 1 dependencies of derivation '/nix/store/xyng04zsf2llhi7kqyc5cmk6vmy1brg9-python3-3.9.12-env.drv' failed to build
```

**Comment by risicle** [comment]
What is the system in question?

**Comment by Mindavi** [comment]
Maybe aarch64-darwin, like siraben? (note that that's also a list of only failures)

**Comment by risicle** [comment]
If it is darwin, is it possible you might have been building something else at the same time that caused a local port number collision?

**Comment by alexvorobiev** [comment]
It is Linux. Multi-user nix on RHEL.

**Comment by risicle** [comment]
x86_64?

**Comment by SuperSandro2000** [comment]
Works for me on x86_64-linux

**Comment by risicle** [comment]
It would be useful to see the content of that truncated error message `FAILED gdrivers/wcs.py::test_wcs_3 - AttributeError: module 'gdaltest' has no...`

**Comment by alexvorobiev** [comment]
@risicle How do I get the full message

**Comment by Mindavi** [comment]
`nix build -L -f default.nix gdal`. The `-L` flag enables logging. Also a `nix log /path/to/drv` is normally shown when a build fails (with the experimental nix command line tools)

**Pull request merged**